### PR TITLE
Return 403 Response when user is not authorized to list organizations

### LINKED ISSUE #2990: Server error at /organization if not authorized to list organizations
### CKAN Version if known (or site URL)
2.5.2

### Please describe the expected behaviour
The behaviour should be similar to if the user goes to /organization/foo and does not have "organization_show" access ie the user gets logged out and sees a message "Organizations are not available."

### Please describe the actual behaviour
Server error:
```
URL: http://localhost:5000/organization
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/weberror/evalexception.py', line 428 in respond
  app_iter = self.application(environ, detect_start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/dec.py', line 147 in __call__
  resp = self.call_func(req, *args, **self.kwargs)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/dec.py', line 208 in call_func
  return self.func(req, *args, **kwargs)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/fanstatic/publisher.py', line 234 in __call__
  return request.get_response(self.app)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/request.py', line 1053 in get_response
  application, catch_exc_info=False)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/request.py', line 1022 in call_application
  app_iter = application(self.environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/dec.py', line 147 in __call__
  resp = self.call_func(req, *args, **self.kwargs)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/dec.py', line 208 in call_func
  return self.func(req, *args, **kwargs)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/fanstatic/injector.py', line 54 in __call__
  response = request.get_response(self.app)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/request.py', line 1053 in get_response
  application, catch_exc_info=False)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/webob/request.py', line 1022 in call_application
  app_iter = application(self.environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/config/middleware.py', line 389 in inner
  result = application(environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/beaker/middleware.py', line 73 in __call__
  return self.app(environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/beaker/middleware.py', line 155 in __call__
  return self.wrap_app(environ, session_start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/routes/middleware.py', line 131 in __call__
  response = self.app(environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/wsgiapp.py', line 125 in __call__
  response = self.dispatch(controller, environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/wsgiapp.py', line 324 in dispatch
  return controller(environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/lib/base.py', line 337 in __call__
  res = WSGIController.__call__(self, environ, start_response)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/controllers/core.py', line 221 in __call__
  response = self._dispatch_call()
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/controllers/core.py', line 172 in _dispatch_call
  response = self._inspect_call(func)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/controllers/core.py', line 107 in _inspect_call
  result = self._perform_call(func, args)
File '/home/martinb/workspace/ckan/lib/default/local/lib/python2.7/site-packages/pylons/controllers/core.py', line 60 in _perform_call
  return func(**args)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/controllers/group.py', line 178 in index
  data_dict_global_results)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/logic/__init__.py', line 416 in wrapped
  result = _action(context, data_dict, **kw)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/logic/action/get.py', line 608 in organization_list
  _check_access('organization_list', context, data_dict)
File '/home/martinb/workspace/ckan/lib/default/src/ckan/ckan/logic/__init__.py', line 286 in check_access
  raise NotAuthorized(msg)
NotAuthorized: <function unauthorized at 0x7f1d7e787410> requires an authenticated user
```

### What steps can be taken to reproduce the issue? 
Log in with a user that does not have "organization_list" access and go to /organization

### PR DESCRIPTION
**Pull request by k-nut**
Fixes #2990 

### Proposed fixes:
When a user tries to access `/organization` now but does not have permission for `organization_list` a 403 response is returned instead of the backend breaking because of an uncaught exception of type `NotAuthorized`

### Features:

- [x] includes tests covering changes
- [ ] includes updated documentation
- [x] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

### CODE REVIEW

File: ckan/logic/action/get.py [hunk 1]
```diff
@@ -518,10 +519,13 @@ def organization_list(context, data_dict):
...
-    data_dict['groups'] = data_dict.pop('organizations', [])
-    data_dict.setdefault('type', 'organization')
-    return _group_or_org_list(context, data_dict, is_org=True)
+    try:
+        _check_access('organization_list', context, data_dict)
+        data_dict['groups'] = data_dict.pop('organizations', [])
+        data_dict.setdefault('type', 'organization')
+        return _group_or_org_list(context, data_dict, is_org=True)
+    except logic.NotAuthorized:
+        base.abort(403, _('You are not authorized to see a list of organizations'))
```
Comment (wardi) [inline ckan/logic/action/get.py]: aborting is something controllers do. I think it's correct for this logic function to just raise the NotAuthorized exception from the _check_access call. The problem is one of the controllers that calls this action isn't handling NotAuthorized.

Comment (k-nut) [inline ckan/logic/action/get.py]: That makes sense. I like the change I am pushing now much better ;)

**Comment by k-nut** [comment]
I just moved the check for permission to the controller. I am not sure if the line above (` self._check_access('site_read', context)`) is still needed since apparently [no one knows exactly what that is doing](https://github.com/ckan/ckan/blob/master/ckan/logic/auth/get.py#L15)

**Comment by wardi** [comment]
@k-nut site_read could be used to limit access to the whole site until you've logged in. This isn't a common thing for open data portals to want, so it hasn't had any work or testing put into it. I can understand wanting to restrict access this way when using ckan for other types of data, though.

**Comment by k-nut** [comment]
@wardi ah okay, that makes sense. So we can just leave it in there next to the new check :)

**Pull request merged**